### PR TITLE
Licensing Portal: implement the unassign licenses feature

### PR DESCRIPTION
**Pull request by wilmersondasilva**
#### Proposed Changes

* Licensing Portal: add the possibility to unassign a license from a site. Thus the users can unlinking a license from a specific site and reuse it on another site.

#### Testing Instructions

* Make sure you have a wordpress.com website.
* Open up Licensing -> Licenses.
* Issue a new license, if don't have one.
* Open some license details. If this license is not assigned to some site, please do it.
* The "Unassign License" button should appear on the bottom left corner.
* Click on the "Unassign License" button. An alert modal should appear with information about this action and ask for you to confirm that want to unassign that license from the site.

#### Screenshots
image

image

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React, or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have you sent any new strings for translation (PCYsg-1vr-p2) ASAP?

Related to #

### CODE REVIEW

File: client/jetpack-cloud/sections/partner-portal/unassign-license-dialog/style.scss [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		align-items: center;
+		background: var( --studio-gray-0 );
+		border: 0;
+
+		a {
+			margin-right: auto;
+			font-weight: 600;
+			text-decoration: underline;
+		}
+	}
```
Comment (heydemoura) [inline client/jetpack-cloud/sections/partner-portal/unassign-license-dialog/style.scss:25]: These styles are a little outdated. Our current production version has a different style for the dialog components on small screens.

Such as this revoke dialog:
image

I would suggest rebasing this branch and applying the updated styles from the revoke dialog to this dialog as well.

Making those styles to be shared somehow would be nice, maybe having a higher Dialog component to apply these shared styles.

Comment (heydemoura) [reply client/jetpack-cloud/sections/partner-portal/unassign-license-dialog/style.scss:25]: We could skip the creation of a higher Dialog component for now and have it as a maintenance task to be looked at in the future, as we currently just have these 2 dialogs.

**Review by heydemoura — CHANGES_REQUESTED** [review]
Code is looking good, just need some rebasing and updating some styles with updated versions from `trunk`.

**Comment by wilmersondasilva** [comment]
I closed this PR because a rebase operation went wrong, and it added a lot of commits that were not supposed to be in this PR. I'll create a new one containing only the correct commits. I'm sorry for those who got requested to review this PR. It happened automatically. I'm sorry!

**Pull request closed without merging**